lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -339,11 +339,8 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
         if self.service_principal_client_id or self.service_principal_key:
             add_secret(self.service_principal_client_id, mask=PATTERN_GUID)
             add_secret(self.service_principal_key)
-            if not self.service_principal_client_id or not self.service_principal_key:
-                raise LisaException(
-                    "service_principal_client_id and service_principal_key "
-                    "should be specified either both or not."
-                )
+            if not self.service_principal_client_id:
```

Comment:
@LiliDeng, with supports of managed identity, even the client id should be optional, is it right?

Reply:
with msi, yes, no need client id, if use azure cli login, client id is still needed

Reply:
If the client id isn't required in all cases, it should be optional.